invenio_pidstore/providers/recordid_v2.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015-2019 CERN.
+# Copyright (C) 2019 Northwestern University.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""DataCite PID provider."""
+
+from __future__ import absolute_import
+
+import copy
+
+from base32_lib import base32
+from flask import current_app
+
+from ..models import PIDStatus
+from .base import BaseProvider
+
+
+class RecordIdProviderV2(BaseProvider):
+    """Record identifier provider V2.
+
+    This is the recommended record id provider.
+
+    It generates a random alphanumeric string as opposed to an increasing
+    integer (:class:`invenio_pidstore.providers.recordid.RecordIdProvider`).
+    """
+
+    pid_type = 'recid'
```

Comment:
Is this a conscious decision? I think it makes. The value should either be ``recid`` or ``recidv2``. If it's ``recid`` PIDs generated with this provider uses the same namespace as the integers from v1. This impacts e.g. the PID resolver, so essentially you'll be able to mix both e.g. ``/records/1`` and ``/records/3sbk2-5j060``. If you name it ``recidv2`` you should only be able to do ``/records/1`` and ``/anotherrecords/3sbk2-5j060``. I think being able to use both, will make it easier to migrate e.g. Zenodo and other older Invenio services.

Reply:
Per chat, this is on purpose.